File: schemaorg-rs/src/menu_item.rs
use crate::*;
use serde_with::{serde_as, OneOrMany};
///<https://schema.org/additionalType>
///<https://schema.org/Text>
///<https://schema.org/URL>
pub type MenuItemAdditionalTypeFieldEnum = String;
///<https://schema.org/MenuItem>
#[serde_as]
#[derive(Debug, serde::Deserialize)]
#[cfg_attr(feature = "uniffi", derive(uniffi::Record))]
pub struct MenuItem {
    #[serde(rename = "@context")]
    pub context: String,
    ///<https://schema.org/url>
    #[serde(rename = "url")]
    #[serde_as(as = "OneOrMany<_>")]
    pub url: Vec<String>,
    ///<https://schema.org/image>
    #[serde(rename = "image")]
    #[serde_as(as = "OneOrMany<_>")]
    pub image: Vec<MenuItemImageFieldEnum>,
    ///<https://schema.org/description>
    #[serde(rename = "description")]
    #[serde_as(as = "OneOrMany<_>")]
    pub description: Vec<MenuItemDescriptionFieldEnum>,
    ///<https://schema.org/suitableForDiet>
    #[serde(rename = "suitableForDiet")]
    #[serde_as(as = "OneOrMany<_>")]
    pub suitable_for_diet: Vec<RestrictedDietEnum>,
    ///<https://schema.org/potentialAction>
    #[serde(rename = "potentialAction")]
    #[serde_as(as = "OneOrMany<_>")]
    pub potential_action: Vec<Action>,
    ///<https://schema.org/sameAs>
    #[serde(rename = "sameAs")]
    #[serde_as(as = "OneOrMany<_>")]
    pub same_as: Vec<String>,
    ///<https://schema.org/mainEntityOfPage>
    #[serde(rename = "mainEntityOfPage")]
    #[serde_as(as = "OneOrMany<_>")]
    pub main_entity_of_page: Vec<MenuItemMainEntityOfPageFieldEnum>,
    ///<https://schema.org/menuAddOn>
    #[serde(rename = "menuAddOn")]
    #[serde_as(as = "OneOrMany<_>")]
    pub menu_add_on: Vec<MenuItemMenuAddOnFieldEnum>,
    ///<https://schema.org/disambiguatingDescription>
    #[serde(rename = "disambiguatingDescription")]
    #[serde_as(as = "OneOrMany<_>")]
    pub disambiguating_description: Vec<String>,
    ///<https://schema.org/identifier>
    #[serde(rename = "identifier")]
    #[serde_as(as = "OneOrMany<_>")]
    pub identifier: Vec<MenuItemIdentifierFieldEnum>,
    ///<https://schema.org/additionalType>
    #[serde(rename = "additionalType")]
    #[serde_as(as = "OneOrMany<_>")]
    pub additional_type: Vec<MenuItemAdditionalTypeFieldEnum>,
    ///<https://schema.org/name>
    #[serde(rename = "name")]
    #[serde_as(as = "OneOrMany<_>")]
    pub name: Vec<String>,
    ///<https://schema.org/offers>
    #[serde(rename = "offers")]
    #[serde_as(as = "OneOrMany<_>")]
    pub offers: Vec<MenuItemOffersFieldEnum>,
    ///<https://schema.org/nutrition>
    #[serde(rename = "nutrition")]
    #[serde_as(as = "OneOrMany<_>")]
    pub nutrition: Vec<NutritionInformation>,
    ///<https://schema.org/alternateName>
    #[serde(rename = "alternateName")]
    #[serde_as(as = "OneOrMany<_>")]
    pub alternate_name: Vec<String>,
    ///<https://schema.org/subjectOf>
    #[serde(rename = "subjectOf")]
    #[serde_as(as = "OneOrMany<_>")]
    pub subject_of: Vec<MenuItemSubjectOfFieldEnum>,
}

File: schemaorg-ts/src/menu_item.ts
import type * as all from "./index";

/**
 * <https://schema.org/additionalType>
 **/
export type MenuItemAdditionalTypeFieldEnum = string;

/**
 * <https://schema.org/MenuItem>
 **/
export interface MenuItem {
  "@context": string;
  "@type": "MenuItem";

  /**
   * <https://schema.org/offers>
   **/
  offers: all.OneOrMany<all.MenuItemOffersFieldEnum>;

  /**
   * <https://schema.org/suitableForDiet>
   **/
  suitableForDiet: all.OneOrMany<all.RestrictedDietEnum>;

  /**
   * <https://schema.org/nutrition>
   **/
  nutrition: all.OneOrMany<all.NutritionInformation>;

  /**
   * <https://schema.org/menuAddOn>
   **/
  menuAddOn: all.OneOrMany<all.MenuItemMenuAddOnFieldEnum>;

  /**
   * <https://schema.org/name>
   **/
  name: all.OneOrMany<string>;

  /**
   * <https://schema.org/mainEntityOfPage>
   **/
  mainEntityOfPage: all.OneOrMany<all.MenuItemMainEntityOfPageFieldEnum>;

  /**
   * <https://schema.org/url>
   **/
  url: all.OneOrMany<string>;

  /**
   * <https://schema.org/disambiguatingDescription>
   **/
  disambiguatingDescription: all.OneOrMany<string>;

  /**
   * <https://schema.org/identifier>
   **/
  identifier: all.OneOrMany<all.MenuItemIdentifierFieldEnum>;

  /**
   * <https://schema.org/description>
   **/
  description: all.OneOrMany<all.MenuItemDescriptionFieldEnum>;

  /**
   * <https://schema.org/subjectOf>
   **/
  subjectOf: all.OneOrMany<all.MenuItemSubjectOfFieldEnum>;

  /**
   * <https://schema.org/alternateName>
   **/
  alternateName: all.OneOrMany<string>;

  /**
   * <https://schema.org/additionalType>
   **/
  additionalType: all.OneOrMany<all.MenuItemAdditionalTypeFieldEnum>;

  /**
   * <https://schema.org/potentialAction>
   **/
  potentialAction: all.OneOrMany<all.Action>;

  /**
   * <https://schema.org/sameAs>
   **/
  sameAs: all.OneOrMany<string>;

  /**
   * <https://schema.org/image>
   **/
  image: all.OneOrMany<all.MenuItemImageFieldEnum>;
}
